spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -235,7 +235,7 @@ protected String createDeploymentId(AppDeploymentRequest request) {
 	}
 
 
-	private void launch(String appId, AppDeploymentRequest request) {
+	private synchronized void launch(String appId, AppDeploymentRequest request) {
```

Comment:
I see that you sync here rather than the `public void launch(AppDeploymentRequest)` method. 

Do you mind explaining how the race condition was occurring?

Reply:
Great question. When remote partitioning a batch job you can specify a Executor like ThreadPoolTaskExecutor and thus the partitions are launched asynchronously and each partition launches a pod using the same instance of KubernetesTaskLauncher. What I noticed is that the same pod name was being re-used on some of the launch requests and thus an exception was thrown from Fabrik8 saying pod with x name already exists.

Reply:
>  What I noticed is that the same pod name was being re-used on some of the launch requests 

Got it. I am trying to identify where in the code this is happening.